Does not compile against latest ouroboros-consensus

When building against the latest master in `ouroboros-consensus`:

```
[ 7 of 10] Compiling TxSubmission

/home/nc/topos/iohk/cardano-node/app/TxSubmission.hs:228:27: error:
    • Could not deduce: Block.HeaderHash blk ~ Block.ChainHash block0
      from the context: (RunDemo blk, MonadST m)
        bound by the type signature for:
                   localChainSyncCodec :: forall blk (m :: * -> *).
                                          (RunDemo blk, MonadST m) =>
                                          NodeConfig (BlockProtocol blk)
                                          -> Codec
                                               (ChainSync blk (Point blk))
                                               DeserialiseFailure
                                               m
                                               ByteString
        at app/TxSubmission.hs:(219,1)-(223,42)
      Expected type: Block.HeaderHash blk -> Codec.CBOR.Encoding.Encoding
        Actual type: Block.ChainHash block0
                     -> Codec.CBOR.Encoding.Encoding
      The type variable ‘block0’ is ambiguous
    • In the first argument of ‘Block.encodePoint’, namely
        ‘(Block.encodeChainHash demoEncodeHeaderHash)’
      In the third argument of ‘codecChainSync’, namely
        ‘(Block.encodePoint (Block.encodeChainHash demoEncodeHeaderHash))’
      In the expression:
        codecChainSync
          (demoEncodeBlock pInfoConfig)
          (demoDecodeBlock pInfoConfig)
          (Block.encodePoint (Block.encodeChainHash demoEncodeHeaderHash))
          (Block.decodePoint (Block.decodeChainHash demoDecodeHeaderHash))
    • Relevant bindings include
        pInfoConfig :: NodeConfig (BlockProtocol blk)
          (bound at app/TxSubmission.hs:224:21)
        localChainSyncCodec :: NodeConfig (BlockProtocol blk)
                               -> Codec
                                    (ChainSync blk (Point blk)) DeserialiseFailure m ByteString
          (bound at app/TxSubmission.hs:224:1)
    |
228 |       (Block.encodePoint (Block.encodeChainHash demoEncodeHeaderHash))
    |                           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```

--- a/app/Run.hs
+++ b/app/Run.hs
@@ -39,7 +39,7 @@ import           Cardano.BM.Trace (Trace, appendName)
 import qualified Ouroboros.Network.AnchoredFragment as AF
 import           Ouroboros.Network.Block
 import qualified Ouroboros.Network.Block as Block
-import           Ouroboros.Network.Chain (genesisPoint, pointHash)
+import           Ouroboros.Network.Chain (genesisPoint)
 import qualified Ouroboros.Network.Chain as Chain
 import           Ouroboros.Network.NodeToClient as NodeToClient
 import           Ouroboros.Network.NodeToNode as NodeToNode
@@ -318,11 +318,11 @@ handleSimpleNode p NodeCLIArguments{..} myNodeAddress (TopologyInfo myNodeId top
 
       encodePoint' ::  Point blk -> Encoding
       encodePoint' =
-          Block.encodePoint $ Block.encodeChainHash demoEncodeHeaderHash
+          Block.encodePoint demoEncodeHeaderHash
 
       decodePoint' :: forall s. Decoder s (Point blk)
       decodePoint' =
-          Block.decodePoint $ Block.decodeChainHash demoDecodeHeaderHash
+          Block.decodePoint demoDecodeHeaderHash
 
 
 removeStaleLocalSocket :: FilePath -> IO ()

--- a/app/TxSubmission.hs
+++ b/app/TxSubmission.hs
@@ -225,8 +225,8 @@ localChainSyncCodec pInfoConfig =
     codecChainSync
       (demoEncodeBlock pInfoConfig)
       (demoDecodeBlock pInfoConfig)
-      (Block.encodePoint (Block.encodeChainHash demoEncodeHeaderHash))
-      (Block.decodePoint (Block.decodeChainHash demoDecodeHeaderHash))
+      (Block.encodePoint demoEncodeHeaderHash)
+      (Block.decodePoint demoDecodeHeaderHash)
 
 
 localSocketFilePath :: NodeId -> FilePath

--- a/cabal.project
+++ b/cabal.project
@@ -86,37 +86,37 @@ source-repository-package
 source-repository-package
   type: git
   location: https://github.com/input-output-hk/ouroboros-network
-  tag: f132a47c7d8711cc2c0457c62509fa1dbcb9347c
+  tag: f3f5f4903d88bcbd52496c38046732b22515cec2
   subdir: ouroboros-network
 
 source-repository-package
   type: git
   location: https://github.com/input-output-hk/ouroboros-network
-  tag: f132a47c7d8711cc2c0457c62509fa1dbcb9347c
+  tag: f3f5f4903d88bcbd52496c38046732b22515cec2
   subdir: ouroboros-consensus
 
 source-repository-package
   type: git
   location: https://github.com/input-output-hk/ouroboros-network
-  tag: f132a47c7d8711cc2c0457c62509fa1dbcb9347c
+  tag: f3f5f4903d88bcbd52496c38046732b22515cec2
   subdir: typed-protocols
 
 source-repository-package
   type: git
   location: https://github.com/input-output-hk/ouroboros-network
-  tag: f132a47c7d8711cc2c0457c62509fa1dbcb9347c
+  tag: f3f5f4903d88bcbd52496c38046732b22515cec2
   subdir: typed-protocols-cbor
 
 source-repository-package
   type: git
   location: https://github.com/input-output-hk/ouroboros-network
-  tag: f132a47c7d8711cc2c0457c62509fa1dbcb9347c
+  tag: f3f5f4903d88bcbd52496c38046732b22515cec2
   subdir: network-mux
 
 source-repository-package
   type: git
   location: https://github.com/input-output-hk/ouroboros-network
-  tag: f132a47c7d8711cc2c0457c62509fa1dbcb9347c
+  tag: f3f5f4903d88bcbd52496c38046732b22515cec2
   subdir: io-sim-classes
 
 source-repository-package

--- a/cardano-node/Cardano/Node/CLI.hs
+++ b/cardano-node/Cardano/Node/CLI.hs
@@ -1,4 +1,5 @@
 {-# LANGUAGE GADTs #-}
+{-# LANGUAGE FlexibleContexts #-}
 
 module Cardano.Node.CLI (
   -- * Untyped/typed protocol boundary

--- a/nix/.stack.nix/io-sim-classes.nix
+++ b/nix/.stack.nix/io-sim-classes.nix
@@ -4,8 +4,8 @@
     package = {
       specVersion = "1.10";
       identifier = { name = "io-sim-classes"; version = "0.1.0.0"; };
-      license = "MIT";
-      copyright = "2018 IOHK";
+      license = "Apache-2.0";
+      copyright = "";
       maintainer = "";
       author = "Alexander Vieth, Marcin Szamotulski, Duncan Coutts";
       homepage = "";
@@ -29,8 +29,8 @@
     } // {
     src = (pkgs.lib).mkDefault (pkgs.fetchgit {
       url = "https://github.com/input-output-hk/ouroboros-network";
-      rev = "f132a47c7d8711cc2c0457c62509fa1dbcb9347c";
-      sha256 = "1vp0m4zhv93z7x4vwaj5izp13dzx5s44wa6q0dpkq0l8x73pi47x";
+      rev = "f3f5f4903d88bcbd52496c38046732b22515cec2";
+      sha256 = "0p2n3dali5cy3s2fzb8h1f6n8hh5pdqpq9mkj39rhf07s59kf05w";
       });
     postUnpack = "sourceRoot+=/io-sim-classes; echo source root reset to \$sourceRoot";
     }

--- a/nix/.stack.nix/network-mux.nix
+++ b/nix/.stack.nix/network-mux.nix
@@ -4,8 +4,8 @@
     package = {
       specVersion = "1.10";
       identifier = { name = "network-mux"; version = "0.1.0.0"; };
-      license = "MIT";
-      copyright = "2019 IOHK";
+      license = "Apache-2.0";
+      copyright = "";
       maintainer = "duncan@well-typed.com, marcin.szamotulski@iohk.io, marc.fontaine@iohk.io, karl.knutsson@iohk.io, alex@well-typed.com";
       author = "Duncan Coutts, Marc Fontaine, Karl Knutsson, Marcin Szamotulski, Alexander Vieth";
       homepage = "";
@@ -62,8 +62,8 @@
     } // {
     src = (pkgs.lib).mkDefault (pkgs.fetchgit {
       url = "https://github.com/input-output-hk/ouroboros-network";
-      rev = "f132a47c7d8711cc2c0457c62509fa1dbcb9347c";
-      sha256 = "1vp0m4zhv93z7x4vwaj5izp13dzx5s44wa6q0dpkq0l8x73pi47x";
+      rev = "f3f5f4903d88bcbd52496c38046732b22515cec2";
+      sha256 = "0p2n3dali5cy3s2fzb8h1f6n8hh5pdqpq9mkj39rhf07s59kf05w";
       });
     postUnpack = "sourceRoot+=/network-mux; echo source root reset to \$sourceRoot";
     }

--- a/nix/.stack.nix/ouroboros-consensus.nix
+++ b/nix/.stack.nix/ouroboros-consensus.nix
@@ -4,8 +4,8 @@
     package = {
       specVersion = "1.10";
       identifier = { name = "ouroboros-consensus"; version = "0.1.0.0"; };
-      license = "MIT";
-      copyright = "2018 IOHK";
+      license = "Apache-2.0";
+      copyright = "";
       maintainer = "operations@iohk.io";
       author = "IOHK Engineering Team";
       homepage = "";
@@ -120,8 +120,8 @@
     } // {
     src = (pkgs.lib).mkDefault (pkgs.fetchgit {
       url = "https://github.com/input-output-hk/ouroboros-network";
-      rev = "f132a47c7d8711cc2c0457c62509fa1dbcb9347c";
-      sha256 = "1vp0m4zhv93z7x4vwaj5izp13dzx5s44wa6q0dpkq0l8x73pi47x";
+      rev = "f3f5f4903d88bcbd52496c38046732b22515cec2";
+      sha256 = "0p2n3dali5cy3s2fzb8h1f6n8hh5pdqpq9mkj39rhf07s59kf05w";
       });
     postUnpack = "sourceRoot+=/ouroboros-consensus; echo source root reset to \$sourceRoot";
     }

--- a/nix/.stack.nix/ouroboros-network.nix
+++ b/nix/.stack.nix/ouroboros-network.nix
@@ -4,8 +4,8 @@
     package = {
       specVersion = "1.10";
       identifier = { name = "ouroboros-network"; version = "0.1.0.0"; };
-      license = "MIT";
-      copyright = "2018 IOHK";
+      license = "Apache-2.0";
+      copyright = "";
       maintainer = "";
       author = "Alexander Vieth, Marcin Szamotulski, Duncan Coutts";
       homepage = "";
@@ -115,8 +115,8 @@
     } // {
     src = (pkgs.lib).mkDefault (pkgs.fetchgit {
       url = "https://github.com/input-output-hk/ouroboros-network";
-      rev = "f132a47c7d8711cc2c0457c62509fa1dbcb9347c";
-      sha256 = "1vp0m4zhv93z7x4vwaj5izp13dzx5s44wa6q0dpkq0l8x73pi47x";
+      rev = "f3f5f4903d88bcbd52496c38046732b22515cec2";
+      sha256 = "0p2n3dali5cy3s2fzb8h1f6n8hh5pdqpq9mkj39rhf07s59kf05w";
       });
     postUnpack = "sourceRoot+=/ouroboros-network; echo source root reset to \$sourceRoot";
     }

--- a/nix/.stack.nix/typed-protocols-cbor.nix
+++ b/nix/.stack.nix/typed-protocols-cbor.nix
@@ -4,7 +4,7 @@
     package = {
       specVersion = "1.10";
       identifier = { name = "typed-protocols-cbor"; version = "0.1.0.0"; };
-      license = "MIT";
+      license = "Apache-2.0";
       copyright = "";
       maintainer = "alex@well-typed.com, duncan@well-typed.com, marcin.szamotulski@iohk.io";
       author = "Alexander Vieth, Duncan Coutts, Marcin Szamotulski";
@@ -44,8 +44,8 @@
     } // {
     src = (pkgs.lib).mkDefault (pkgs.fetchgit {
       url = "https://github.com/input-output-hk/ouroboros-network";
-      rev = "f132a47c7d8711cc2c0457c62509fa1dbcb9347c";
-      sha256 = "1vp0m4zhv93z7x4vwaj5izp13dzx5s44wa6q0dpkq0l8x73pi47x";
+      rev = "f3f5f4903d88bcbd52496c38046732b22515cec2";
+      sha256 = "0p2n3dali5cy3s2fzb8h1f6n8hh5pdqpq9mkj39rhf07s59kf05w";
       });
     postUnpack = "sourceRoot+=/typed-protocols-cbor; echo source root reset to \$sourceRoot";
     }

--- a/nix/.stack.nix/typed-protocols.nix
+++ b/nix/.stack.nix/typed-protocols.nix
@@ -4,8 +4,8 @@
     package = {
       specVersion = "1.10";
       identifier = { name = "typed-protocols"; version = "0.1.0.0"; };
-      license = "MIT";
-      copyright = "2018-2019 IOHK";
+      license = "Apache-2.0";
+      copyright = "";
       maintainer = "alex@well-typed.com, duncan@well-typed.com, marcin.szamotulski@iohk.io";
       author = "Alexander Vieth, Duncan Coutts, Marcin Szamotulski";
       homepage = "";
@@ -41,8 +41,8 @@
     } // {
     src = (pkgs.lib).mkDefault (pkgs.fetchgit {
       url = "https://github.com/input-output-hk/ouroboros-network";
-      rev = "f132a47c7d8711cc2c0457c62509fa1dbcb9347c";
-      sha256 = "1vp0m4zhv93z7x4vwaj5izp13dzx5s44wa6q0dpkq0l8x73pi47x";
+      rev = "f3f5f4903d88bcbd52496c38046732b22515cec2";
+      sha256 = "0p2n3dali5cy3s2fzb8h1f6n8hh5pdqpq9mkj39rhf07s59kf05w";
       });
     postUnpack = "sourceRoot+=/typed-protocols; echo source root reset to \$sourceRoot";
     }

--- a/stack.yaml
+++ b/stack.yaml
@@ -54,7 +54,7 @@ extra-deps:
     #Ouroboros-network dependencies
 
   - git: https://github.com/input-output-hk/ouroboros-network
-    commit: f132a47c7d8711cc2c0457c62509fa1dbcb9347c
+    commit: f3f5f4903d88bcbd52496c38046732b22515cec2
     subdirs:
         - io-sim-classes
         - network-mux

--- a/wallet-client/WalletClient.hs
+++ b/wallet-client/WalletClient.hs
@@ -81,11 +81,11 @@ muxLocalInitiatorNetworkApplication
   -- from 'cardano-chain'.  This should remove the dependency of this module
   -- from 'ouroboros-consensus'.
   => Proxy blk
-  -> Tracer m (TraceSendRecv (ChainSync blk (Point blk)))
+  -> Tracer m (TraceSendRecv (ChainSync blk (Point blk)) DeserialiseFailure)
   -- ^ tracer which logs all chain-sync messages send and received by the client
   -- (see 'Ouroboros.Network.Protocol.ChainSync.Type' in 'ouroboros-network'
   -- package)
-  -> Tracer m (TraceSendRecv (LocalTxSubmission (GenTx blk) String))
+  -> Tracer m (TraceSendRecv (LocalTxSubmission (GenTx blk) String) DeserialiseFailure)
   -- ^ tracer which logs all local tx submission protocol messages send and
   -- received by the client (see 'Ouroboros.Network.Protocol.LocalTxSubmission.Type'
   -- in 'ouroboros-network' package).
@@ -196,8 +196,8 @@ localChainSyncCodec pInfoConfig =
     codecChainSync
       (demoEncodeBlock pInfoConfig)
       (demoDecodeBlock pInfoConfig)
-      (Block.encodePoint (Block.encodeChainHash demoEncodeHeaderHash))
-      (Block.decodePoint (Block.decodeChainHash demoDecodeHeaderHash))
+      (Block.encodePoint demoEncodeHeaderHash)
+      (Block.decodePoint demoDecodeHeaderHash)
 
 
 -- | Local unix socket file path over which the client communicates with a core

